### ISSUE TITLE
Asus Gladius II WIRELESS not supported ?

### ISSUE THREAD

**Issue by gletrequess** [seerge/g-helper#2762]
### Mouse / peripheral model and version

Asus Gladius II Wireless

### Armoury Crate Screenshots

![image](https://github.com/seerge/g-helper/assets/113016302/128d6d4a-606f-4a0e-8d82-b1aa34116070)
![image](https://github.com/seerge/g-helper/assets/113016302/c16fd0f2-f9a8-4109-9739-0a007d104dd6)

### USB Details

-

### Additional information.

I would know if Gladius II wireless is supported ? I saw that Gladius II and Gladius Origin was supported for GHelper and Gladius III & Gladius III Wireless, but not Gladius II wireless ?

**Comment by seerge** [comment]
@gletrequess hello, 

Please include USB details from USB Tree Viewer app as it was clearly requested in issue template.

Thanks.

**Comment by gletrequess** [comment]
Hum... wdym USB details ? I'm not really sure what is that sorry !! I mean, maybe it's more a question than a issue, I would just know if Gladius II Wireless is a supported model for Ghelper ?

**Comment by seerge** [comment]
@gletrequess the Peripheral Support request template you have filled had a question about USB details of your device

If you connect your device via dongle and it's not visible in the app, I need to know device details in order to check it.

Also please include screenshots of **all** Armoury Crate pages for this mouse.

Thanks.

> Also, please use [USB Tree View](https://www.uwe-sieber.de/usbtreeview.html) and send the text (especially the block with all the child devices) of the mouse. image
![image](https://github.com/seerge/g-helper/assets/5920850/400b97ef-e8ab-4d8e-90ae-cff86ae4b73a)

**Comment by gletrequess** [comment]
![image](https://github.com/seerge/g-helper/assets/113016302/eac9c7e3-8755-40f0-afa6-1526158daeff)

Okay, I think it's this ? I'm not sure and here all the screenshots from AC
![image](https://github.com/seerge/g-helper/assets/113016302/a271db14-5134-4272-9057-9c2d2120feb6)
![image](https://github.com/seerge/g-helper/assets/113016302/fe57aa67-5412-4a6c-acf2-61ec4569fdf6)
![image](https://github.com/seerge/g-helper/assets/113016302/cef53dba-a1b3-4180-940d-3d79f2cd7e39)
![image](https://github.com/seerge/g-helper/assets/113016302/9443fcdd-e981-4172-a301-5ef5988d9274)
![image](https://github.com/seerge/g-helper/assets/113016302/a7b7fa03-2e59-4d07-a858-b911e4757acf)
![image](https://github.com/seerge/g-helper/assets/113016302/a40c7432-054f-41b0-bdf3-764626bafffa)

I don't really know what is "dongle" ? I used the USB "key" for the mouse maybe it's called a dongle ?

**Comment by seerge** [comment]
@gletrequess ok, great. Your device is actually new for the app so it needs to be added from scratch :)

Can you try this build ?
[GHelper.zip](https://github.com/user-attachments/files/15926523/GHelper.zip)

Thanks

**Comment by gletrequess** [comment]
Okay this seems to work I think !! Thank !! But i don't know why the battery is 4% ? 
![image](https://github.com/seerge/g-helper/assets/113016302/5b4809b5-de53-47ad-8e11-b3e40a96b848)

You're really helping sir realy thanks !!

**Comment by seerge** [comment]
@gletrequess can you check if this build shows battery charge correctly ?

[GHelper.zip](https://github.com/user-attachments/files/15926643/GHelper.zip)

**Comment by gletrequess** [comment]
![20240621_130444](https://github.com/seerge/g-helper/assets/113016302/371e0554-010e-43b8-8103-3447271491c3)
![20240621_130604](https://github.com/seerge/g-helper/assets/113016302/71ff6e39-fa12-4068-866d-442670ff8e9b)
 
 Okay so ! Few news, indeed the RGB seems kinda weird, when I try to only activate the Logo settings, the molette is still active and in my second image, I don't really know why the logo is red but the molette is blue ??

**Comment by gletrequess** [comment]
Update !! I didn't know each settings were for change the color and level of lighting, so I think that's good ? My RGB was off previously

**Comment by gletrequess** [comment]
Oh and another thing, I have the "Led ROG" settings but it doesn't change anything on the mouse, it is normal ?

**Comment by seerge** [comment]
@gletrequess I just copied whole "RGB communication" part from a Weird version of Gladius II (that was already supported by the app).

- Do i understand correctly that battery charge reading is correct now ?
- Does RGB settings work (and get reflected in UI) correctly now ?

**Comment by seerge** [comment]
@gletrequess 

> Oh and another thing, I have the "Led ROG" settings but it doesn't change anything on the mouse, it is normal ?

I think LED ROG is responsible for the "Underglow" part of the mouse, if yours don't have it - I can remove it from the list ... :)

**Comment by gletrequess** [comment]
I thin the battery charge is correct yes, my mouse is 100% battery right now and it shows it on both the same AC and GHelper ! FOr the underglow part, well I have the classic thing for the mouse (i'll show you) but when I try to up the level of the "Led ROG" it doesn't change anything ? Maybe it's not for that and my mouse don't have it !
![20240621_131839](https://github.com/seerge/g-helper/assets/113016302/0e5ffa6b-87be-4696-b2de-673927c361c5)

**Comment by seerge** [comment]
@gletrequess I think your mouse simply doesn't have that part of backlight :)

Try this build w/o that option at all

[GHelper.zip](https://github.com/user-attachments/files/15926836/GHelper.zip)

**Comment by gletrequess** [comment]
That is perfect! Great, I have now just one last question, it concerns the color "DPI", I do not think it is very important in all cases but as much to tell you the problem, I make you a small video so that you understand the problem

https://github.com/seerge/g-helper/assets/113016302/3b877cd1-49fc-4227-a07c-aabca6caf2fb

**Comment by gletrequess** [comment]
The color kinda "reset", i don't know if it's normal ?

**Comment by seerge** [comment]
@gletrequess no, it should remember actual color in mouse settings and then read it back

Can you run this build , do same thing and upload a log from ``%appdata%\GHelper`` here after ?

[GHelper.zip](https://github.com/user-attachments/files/15927015/GHelper.zip)

**Comment by gletrequess** [comment]
The results are the same than the previous build ! Here the log from the appadata
[log.txt](https://github.com/user-attachments/files/15927501/log.txt)

**Comment by seerge** [comment]
@gletrequess ok great,

May be @IceStormNG knows why DPI profile colors are not stored here (as I'm not very familiar with the protocol). And it seems like other mice all are supposed to act same at least in this regard ?

**Comment by gletrequess** [comment]
Maybe !! I'll wait for his response then, thanks for your supporrt !

**Comment by IceStormNG** [comment]
The reason is that not all mice support it. Only few do. All other mice will always show the default colors for the 4 DPI profiles.

**Comment by seerge** [comment]
@IceStormNG @gletrequess 

Ok, I have just set ``HasDPIColors`` to OFF completely, try this build instead 

[GHelper.zip](https://github.com/user-attachments/files/15927896/GHelper.zip)

**Comment by gletrequess** [comment]
Okay ! That's seem to be good for me ! I can't change the color anymore ^^

Eeeemm... oh yeah, I just have a last question, when a update will come out, if I update my GHelper with the public release and all, do my mouse will still be supported ? And the settings you've made here still be the same ?

**Comment by seerge** [comment]
@gletrequess if everything is working fine now (right?), I will just include this changes in the next release.

**Comment by gletrequess** [comment]
Yup ! For me everything is working !!

**Comment by seerge** [comment]
Glad to hear, closing as completed then.
